README.md

```diff
@@ -1,175 +1,404 @@
-- **Schema Exploration**: Explore database tables, columns, and relationships
 
 ## Quick Start
 
-#### Using Docker
+### Prerequisites
+
+Before getting started, ensure you have:
+1. Access credentials for your database.
+2. Python 3.12 or higher *or* Docker.
+3. The `pg_statements` and `hypopg` extensions loaded on your database (for full functionality).
+
+#### Access Credentials
+ You can confirm your access credentials are valid by using `psql` or a GUI tool such as [pgAdmin](https://www.pgadmin.org/).
+
+
+#### Python  or Docker
+
+The choice to use Docker or Python is yours.
+We generally recommend using whichever is most familiar to you.
+
+#### Postgres Extensions
+
+Postgres Pro uses the `pg_statements` extension to analyze query execution statistics.
+For example, this allows it to understand which queries are running slow or consuming significant resources.
+It uses the `hypopg` extension to simulate the behavior of the Postgres query planner after adding indexes.
+While Postgres Pro can run without these extensions, its capabilities will be limited.
+
+If your Postgres database is running on a cloud provider managed service, the `pg_statements` and `hypopg` extensions will probably already be available on the system.
+In this case, you can just run `CREATE EXTENSION` commands using a role with sufficient privileges.
+
+If you are using a self-managed Postgres database, you may need to do additional work.
+Before loading the `pg_statements` extension you must ensure that it is listed in the `shared_preload_libraries` in the Postgres configuration file.
+The `hypopg` extension may also require additional system-level installation (e.g., via your package manager) because it does not always ship with Postgres.
+
+
+### Installation
+
+Choose one of the following methods to install Postgres Pro:
+
+#### Option 1: Using Python
+
+If you have `pipx` installed you can install Postgres Pro with:
 
 ```bash
-docker pull crystaldba/postgres-mcp
-docker run -it --rm crystaldba/postgres-mcp "postgres://user:password@host:5432/dbname"
+pipx install postgres-mcp
 ```
 
-When using Docker with a localhost database, our image automatically handles the connection:
+Otherwise, install Postgres Pro with `uv`:
 
-- MacOS/Windows: Uses `host.docker.internal` automatically
-- Linux: Uses `172.17.0.1` or the appropriate host address automatically
+```bash
+uv pip install postgres-mcp
+```
 
-Example with automatic localhost remapping:
+If you need to install `uv`, see the [uv installation instructions](https://docs.astral.sh/uv/getting-started/installation/).
+
+#### Option 2: Using Docker
+
+Pull the Postgres Pro MCP server Docker image.
+This image contains all necessary dependencies, providing a reliable way to run Postgres Pro in a variety of environments.
 
 ```bash
-docker run -it --rm crystaldba/postgres-mcp "postgres://user:password@localhost:5432/dbname"
+docker pull crystaldba/postgres-mcp
 ```
 
```

Comment:
We should add examples showing how users can verify their installation is working correctly.